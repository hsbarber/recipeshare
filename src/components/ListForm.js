import React from 'react';
import PropTypes from 'prop-types';

function ListForm({ addItemArray, inputName }) {
  // Input tracker
  let input1;
  return (
    <div className={`${inputName}s-input`}>
      {inputName === 'ingredient' ? (
        <input
          ref={node => {
            input1 = node;
          }}
        />
      ) : (
        <textarea
          ref={node => {
            input1 = node;
          }}
        />
      )}
      <button
        type="button"
        onClick={e => {
          e.preventDefault();
          addItemArray(input1.value, `${inputName}s`);
          input1.value = '';
        }}
      >
        Add {inputName} +
      </button>
    </div>
  );
}

export default ListForm;

ListForm.propTypes = {
  addItemArray: PropTypes.func,
  inputName: PropTypes.string,
};
