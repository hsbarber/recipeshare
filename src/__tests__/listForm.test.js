import * as React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListForm from '../components/ListForm';
import '../enzyme';

describe('ListForm', () => {
  const addItemArray = jest.fn();
  const inputName = 'ingredient';
  it('shallow renders without crashing', () => {
    shallow(<ListForm addItemArray={addItemArray} inputName={inputName} />);
  });
  it('component matches snapshot', () => {
    const wrapper = shallow(
      <ListForm addItemArray={addItemArray} inputName={inputName} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('addItemArray called on click of add ingredient button', () => {
    const wrapper = mount(
      <ListForm addItemArray={addItemArray} inputName={inputName} />
    );
    wrapper.find('input').instance().value = 'flour';
    wrapper.find('button').simulate('click', {
      preventDefault: () => {},
    });
    expect(addItemArray).toHaveBeenCalled();
  });
});
