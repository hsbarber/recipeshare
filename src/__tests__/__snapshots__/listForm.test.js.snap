// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListForm component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListForm
    addItemArray={[MockFunction]}
    inputName="ingredient"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input />,
        <button
          onClick={[Function]}
          type="button"
        >
          Add 
          ingredient
           +
        </button>,
      ],
      "className": "ingredients-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Add ",
            "ingredient",
            " +",
          ],
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          "Add ",
          "ingredient",
          " +",
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input />,
          <button
            onClick={[Function]}
            type="button"
          >
            Add 
            ingredient
             +
          </button>,
        ],
        "className": "ingredients-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Add ",
              "ingredient",
              " +",
            ],
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            "Add ",
            "ingredient",
            " +",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
